File: api/tracker.py
from flask import Flask, request, send_file
import os

app = Flask(__name__)

@app.route('/tracker')
def tracker():
    user_id = request.args.get('id', 'unknown')
    print(f"[OPENED EMAIL] User ID: {user_id}")

    # Get the absolute path of pixel.png
    image_path = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'pixel.png'))

    if not os.path.exists(image_path):
        print(f"[ERROR] Image not found at: {image_path}")
        return "Image not found", 404

    return send_file(image_path, mimetype='image/png')

# This will let you run locally
if __name__ == "__main__":
    app.run(debug=True)

File: api/tracker.js
const path = require('path');
const fs = require('fs');
const { createClient } = require('@supabase/supabase-js');

const supabaseUrl = process.env.SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_KEY;
const supabase = createClient(supabaseUrl, supabaseKey);

async function saveLogToSupabase(message) {
  const { data, error } = await supabase
    .from('email_logs')
    .insert([{ message }]);

  if (error) {
    console.error('Error saving log to Supabase:', error);
  } else {
    console.log('Log saved to Supabase:', data);
  }
}

module.exports = async (req, res) => {
  const userId = req.query.id || 'unknown';
  const logMessage = `[OPENED EMAIL] User ID: ${userId}`;
  console.log(logMessage);
  await saveLogToSupabase(logMessage);

  // Get the absolute path of pixel.png
  const imagePath = path.resolve(__dirname, '../pixel.png');

  if (!fs.existsSync(imagePath)) {
    const errorMessage = `[ERROR] Image not found at: ${imagePath}`;
    console.error(errorMessage);
    await saveLogToSupabase(errorMessage);
    res.status(404).send('Image not found');
    return;
  }

  res.setHeader('Content-Type', 'image/png');
  res.setHeader('Cache-Control', 'no-store'); // Prevent caching of the pixel
  fs.createReadStream(imagePath).pipe(res);
};